src/modules/launcher/Plugins/Microsoft.Plugin.Program/Programs/Win32Program.cs

```diff
@@ -790,34 +798,40 @@ public bool Equals(Win32 app1, Win32 app2)
             }
 
             // Ref : https://stackoverflow.com/questions/2730865/how-do-i-calculate-a-good-hash-code-for-a-list-of-strings
-            public int GetHashCode(Win32 obj)
+            public int GetHashCode(Win32Program obj)
             {
                 int namePrime = 13;
                 int executablePrime = 17;
                 int fullPathPrime = 31;
 
                 int result = 1;
-                result = result * namePrime + obj.Name.ToLowerInvariant().GetHashCode();
-                result = result * executablePrime + obj.ExecutableName.ToLowerInvariant().GetHashCode();
-                result = result * fullPathPrime + obj.FullPath.ToLowerInvariant().GetHashCode();
+                result = result * namePrime + obj.Name.ToUpperInvariant().GetHashCode(StringComparison.Ordinal);
+                result = result * executablePrime + obj.ExecutableName.ToUpperInvariant().GetHashCode(StringComparison.Ordinal);
+                result = result * fullPathPrime + obj.FullPath.ToUpperInvariant().GetHashCode(StringComparison.Ordinal);

```

Comment:
Any reason for converting the case from ToLower to ToUpper?

Reply:
Yeah this gets flagged by static analysis [CA1308: Normalize strings to uppercase](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1308?view=vs-2019).  

> `A small group of characters, when they are converted to lowercase, cannot make a round trip.`